### DISCUSSION TITLE
Datenpunkt für Sollwerte finden

### DISCUSSION THREAD

**Discussion by platinum83** [open3e/open3e#139] [Q&A] [answered]
Hi, 

meine Vitocal250A läuft nicht ganz so wie sie soll - vermute ich. 

Mit open3e möchte ich feststellen, wie weit die Ist-Werte von den Sollwerten, für die Vorlauftemperatur des Wärmeerzeuges und für die Primärkreispumpe auseinanderliegen. Den Ist-Wert für die Vorlauftemperatur des Wärmeerzeugers habe ich gefunden. Es sollte der Datenpunkt 268 sein.  

Die drei anderen Werte kann ich nicht ausfindig machen. Die vier markierten Werte in den Bildern würde ich gerne auslesen. könnt ihr mir sagen welche Datenpunkte das sein können?

Vielen lieben Dank!

![Vorlauf_Wermeerzeuger](https://github.com/user-attachments/assets/bfca9c44-13f8-4231-b663-8c080c7b146f)
![primaerkreispumpe](https://github.com/user-attachments/assets/1c375b6f-7abf-4a1c-a93e-8e5606e9837e)

**Comment by Wolfgang-03** [comment]
Hallo Platinum83,

987 HK1 Vorlauftemperatur Sollwert
284 HK1 Vorlauftemperatur Istwert
268 Vorlauftemperatur am Wärmeerzeuger
269 Rücklauftemperatur am Wärmeerzeuger

Einen Sollwert am Wärmeerzeuger gibt es bei mir nicht, oder ich kenne ihn nicht. 

2792 HK1 Pumpe Istwert in %

Sollwert für die HK1 Pumpe weiß ich nicht wie der entsteht. Es gibt 1102 mit Min|Max|Setpoint, wenn ich die Werte bei mir verändere hat das keine Auswirkungen auf die Pumpe. Die läuft bei mir immer mit 100%, glaube das liegt bei mir aber an der Pumpe die nur 0 oder 1 kann. 

Gruß
Wolfgang

**Reply by platinum83** [reply]
Hallo Wolfgang,

besten Dank. für den Sollwert der Primärkreispumpe habe ich jetzt einen Eintrag gefunden:

found 381:4:CentralHeatingPump

Das hilft mir aber leider auch nur bedingt, weil ich den Ist-Wert ja nicht kenne. 

Schade dass man an den  Soll-Wert für die Vorlauftemperatur des Wärmeerzeugeses nicht herankommt, das hätte mir weitergeholfen. 

Viele Grüße

**Reply by platinum83** [reply]
ich habe einen Wert 1799 gefunden, dessen Beschreibung vielversprechend aussieht:

open3e -c can0 -r 0x680.[1799] -v
None 1799 PrimaryCircuitCurrentTemperatureSetpoint "3b01"

Allerdings kann ich mit der Ausgabe "3b01" nichts anfangen. Zu einen anderen Zeitpunkt lautet die Ausgabe "0000". Kann man da vielleicht noch etwas umrechnen? Was meint ihr?

Danke!

**Reply by MyHomeMyData** [reply, marked as answer]
Nach der üblichen Umrechnung für Temperaturen wären das 0x013b / 10 = 31,5 °C

**Reply by platinum83** [reply]
funktioniert. Danke!

**Reply by Philip-Wiege** [reply]
@platinum83 Ist das dann jetzt der Vorlauf-Sollwert für den Wärmeerzeuger statt der Heizkreise?

**Reply by platinum83** [reply]
Der Wert 1799 PrimaryCircuitCurrentTemperatureSetpoint ist tatschlich gleich dem Wert aus dem Display Vorlauftemperatur Sollwert. Der 

Wert verhält sich so:
Heizbetrieb: gleich dem Zielwert der Vorlauftemperatur des Heizkreises
Warmwasserbetrieb: gleich der Zieltemperatur des Warmwassers
Keine Wärmeanforderung: 0

Ich finden diesen Wert interessant so beobachten, weil der mir ständig zeigt, was gerade im Moment das Ziel der Anlage ist.

**Reply by platinum83** [reply]
wenn dieser Wert noch direkt als float aus open3e rauskommen würde, wäre das ideal ;-). Danke euch!

**Reply by MyHomeMyData** [reply]
Im Branch develop ist 1799 bereits aktualisiert. Mit  `pip install --upgrade git+https://github.com/open3e/open3e.git@develop` solltest Du auf diese Version (0.3.6) aktualisieren können.

**Comment by Wolfgang-03** [comment]
Ja, den 381 kenne ich auch. Da gibt es auch noch den 2791, welcher bei mir fast die gleiche Werte liefert wie 381.

**Reply by philippoo66** [reply]
>Schade dass man an den Soll-Wert für die Vorlauftemperatur des Wärmeerzeugeses nicht herankommt, das hätte mir weitergeholfen.

Wenn der im Display auftaucht, sollte der sich eigentlich auch irgendwo in den Datenpunkten verstecken... Leider kann ich euch da überhupt nicht helfen, weil ich weder ein E3 Gerät geschweige denn eine Wärmepumpe habe...

**Reply by Philip-Wiege** [reply]
Was ist denn eurer Meinung nach der Unterschied der Werte? Müsste es einen Sollwert für den Erzeuger geben und dann nochmal einen Sollwert pro Heizkreis?

**Reply by platinum83** [reply]
Der Sollwert vom Wärmeerzeugers kann nach meinem Verständnis nicht gleich dem Sollwert vorm HK1 sein, weil sich der Sollwert des Wärmerzeugers schon durch den Betriebszustand ändert, also abhängig davon ob der Heizkreis bedient wird oder Warmwasser bereitet werden soll.
Ich werde jetzt aber trotzdem die "987 HK1 Vorlauftemperatur Sollwert" mit aufzeichnen und auswerten. Ich könnte mir auch gut vorstellen dass bei der Warmwasserbereitung der Sollwert des Wärmeerzeugers auch noch etwas über dem der Zieltemperatur des Warmwassers liegt.  Ich melde mich dazu nochmal, wenn ich einen ganzen Tag aufgezeichnet habe.

Ich hoffe aber weiter drauf eine Möglichkeit zu finden den Sollwert des Vorlauf des Wärmeerzeugers abzurufen/ zu berechnen.

**Comment by Philip-Wiege** [comment]
2791 ist ja die relative Leistung der internen Umwälzpumpe in Prozent. 381 ist der dazu gehörige Sollwert.

Ich verstehe die Regelung so:
- 381 sagt 50% Sollwert
- Das heißt 50% vom Maximaldruck
- Wenn der Heizkreis wenig Strömungswiderstand hat, dreht die Pumpe auch fast genauso hoch
- Wenn der Heizkreis viel Widerstand hat, dreht die Pumpe deutlich langsamer, um den gewünschten Druck zu erreichen

Bei mir ist die Hydraulik des Heizkreises leider sehr schlecht. Daher dreht die Pumpe nur mit 70 Prozent und erreicht schon den Gegendruck, der bei 100 Prozent Sollwert erreicht werden soll. Der Warmwasserkreis hat fast keinen Widerstand, daher dreht die Pumpe fast exakt so schnell wie der Sollwert.

**Reply by platinum83** [reply]
Danke. Das ist ein guter Hinweis. Ich beobachte auch hier die beiden Werte. Jetzt verstehe ich auch auch, warum der ist-Wert größer als der Sollwert bei mir ist. 

Offenbar führt ein zu hoher Druck wie bei dir (und schwächer ausgeprägt bei mir) aber nicht dazu, dass der Heizvorgang unterbrochen wird. Das hatte ich zunächst vermutet. Oder gibt es bei dir viele kurze Heizvorgänge und dafür keine längeren? Bei mir tritt das nämlich auf und ich versuche herauszufinden warum... Ich möchte vermeiden, dass die Wärmepumpe ständig ein und ausgeschaltet wird.

**Reply by philippoo66** [reply]
> Wenn der Heizkreis viel Widerstand hat, dreht die Pumpe deutlich langsamer, um den gewünschten Druck zu erreichen

hm, Umwälzpumpen sind doch so wie Staubsauger (also keine Kolbenpumpe oder so), und wenn man einen Staubsauger zuhält, dreht der höher...

**Reply by Philip-Wiege** [reply]
So wie ich das sehe, ist die Pumpe PWM gesteuert und regelt auf das PWM Signal einen Druck/Förderhöhe ein.

Bei 100% PWM würde meine Umwälzpumpe dann versuchen die spezifizieren 8,x Meter Förderhöhe zu erreichen. Da meine Hydraulik aber viel Gegendruck hat, schafft sie diesen Druck schon mit kleinem Volumenstrom / Drehzahl und muss dafür nur auf 80% Drehzahl hochgehen. Das meldet sie als Istwert zurück.

Ich lasse meine Unwälzpumpe auf Vollgas laufen. Das führt zu 1200 Litern pro Stunde Volumenstrom und maximalem Wärmeübergang. Meiner Meinung nach taktet meine WP dadurch weniger / garnicht, da die Minimalleistung auch bei sehr warmen Fußboden noch abgeführt werden kann. Ich habe nur FBH und keinen Puffer und komme in der Übergangszeit mit 1 bis 3 Starts pro Tag klar inklusive WW.

**Reply by platinum83** [reply]
Ich habe die Werte nun bei mir beobachtet und der ist-Wert (CentralHeatingPumpStatus) ist immer etwas unterhalb dem soll-Wert (CentralHeatingPump) nachdem ich alles gereinigt habe...

**Comment by bomilkar** [comment]
Die Diskussion hat mir auch sehr geholfen zu verstehen, was da passiert.
Gestern wurde zum ersten Mal der Filter in der Außeneinheit gereinigt. Der war sehr zugesetzt, noch mit Schmutz von der Installation, Fermit, Späne, ... . Dadurch ist der Volumenstrom seit Inbetriebnahme kontinuierlich zurückgegangen: von anfänglich 1300 auf unter 800 l/h.
Bei gereinigtem Filter ist der Volumenstrom bei über 1800 l/h.
Als Nebeneffekt höre ich ein recht lautes Brummen von der Inneneinheit. Dieses Brummen wird deutlich leiser, wenn ich den Wert von 1100.2 reduziere, etwa auf unter 80%.
Das was ihr da beschrieben habt scheint wohl zuzutreffen: **wenn der Gegendruck steigt (bei zugesetztem Filter), wird die Drehzahl der Pumpe reduziert, und die Pumpe wird leiser.**

BTW: weiß jemand die DID des Volumenstroms?

Vitocal 250A, in Betrieb seit Oktober 2024.

**Reply by Philip-Wiege** [reply]
Den Volumenstrom kannst du über den 1043 auslesen. Der Wert "Actual" ist der aktuelle Volumenstrom in Litern pro Stunde.

Kleine Nebenbaustelle: Bei uns ist die Hydraulik leider so schlecht, dass meine Pumpe für einen akzeptablen Volumenstrom immer sehr laut ist. Daher plane ich immer noch, zwei Pumpen in Reihe zu schalten, damit jede einzelne weniger Drehzahl braucht...

**Reply by bomilkar** [reply]
Zu deiner Nebenbaustelle: ist der Querschnitt zu gering oder ist die Leitung zu lang?

**Reply by Stoffelhessen** [reply]
> Zu deiner Nebenbaustelle: ist der Querschnitt zu gering oder ist die Leitung zu lang?

Meine Außeneinheit ist ca. 17 Meter im Außenbereich entfernt und innen sind es auch noch mal 5-6 Meter, also habe ich einen Wasserweg hin und zurück von ca. 44 Metern. Mein Volumenstrom beim Abtauen beträgt 2000l/h

**Reply by Philip-Wiege** [reply]
Ich hab zwischen Außeneinheit und Inneneinheit auch 35mm Kupfer. Beim Abtauen komme ich auch auf 2000 Liter / Stunde wenn ich will. Die Rohre zur FBH sind aber zu kleine Mehrschichtverbundrohre mit zu vielen Fittings. Da machste nix.

**Comment by bomilkar** [comment]
Danke, Philip!
Da soll einer drauf kommen: "AllengraSensor".
Aber die abgelesenen Daten stimmen überein.
